src/systems/track_controller/TrackController.cc

```diff
@@ -374,11 +397,26 @@ void TrackController::PreUpdate(
   for (auto& entityPose : this->dataPtr->collisionsWorldPose)
```

Comment:
I just noticed that `collisionsWorldPose` will be empty the first time `PreUpdate` is called which means when poses are retrieved from the cache in the `ComputeSurfaceProperties` callback they will all be the identity pose. Of course, after that the poses will be updated correctly. How about iterating through `trackCollisions` instead since the collision entities are added to that in `RegisterCollision`?

Reply:
Good catch. I changed the loop as you suggested. I also added a protection to `ComputeSurfaceProperties()` which skips the rest of the callback if it finds that the collision has no cached pose (hoping that it will have one in the next iteration). However, I'm not sure if this can ever happen.

Do you also know what will happen in the very first simulation iteration? Will all collisions already have their Pose components filled in? in `worldPose()`, there is a fallback of returning zeros if the component does not exist. That's what we wanted to avoid. So should I check for existence of the component before calling `worldPose()`?

Reply:
> Do you also know what will happen in the very first simulation iteration? Will all collisions already have their Pose components filled in? 

Yes, all collisions should have their `Pose` components filled when they're created based their `<pose>` element in the SDFormat file so `worldPose` should always be available and there's no need to check if the component exists.